backend/src/friend/friend.service.ts

```diff
@@ -1,4 +1,452 @@
-import { Injectable } from '@nestjs/common';
+import {
+  Injectable,
+  Res,
+  BadRequestException,
+  ServiceUnavailableException,
+} from '@nestjs/common';
+import { User } from '@prisma/client';
+import { PrismaService } from 'nestjs-prisma';
+import { Response } from 'express';
 
 @Injectable()
-export class FriendService {}
+export class FriendService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  /*****************************************************************************/
+  /* send friend Request */
+  /*****************************************************************************/
+  async sendFriendRequest(
+    user: User,
+    payload: { userEmail: string },
+    res: Response,
+  ) {
+    // check user/payload data
+    const checkData = await this.checkSenderAndReceiver(user, payload);
+    if (checkData !== '') {
+      throw new BadRequestException(checkData);
```

Comment:
This is illogical to me. If 'checkData' is not empty, it's an error. Maybe something like this:
```
const errorMessage = await this.checkSenderAndReceiver(user, payload);
if (errorMessage) throw new BadRequestException(checkData);
```